go/vt/vtgate/planbuilder/operators/union_merging.go

```diff
@@ -199,8 +202,16 @@ func createMergedUnion(
 			continue
 		}
 		deps = deps.Merge(ctx.SemTable.RecursiveDeps(rae.Expr))
-		ctx.SemTable.CopySemanticInfo(rae.Expr, col)
-		ctx.SemTable.CopySemanticInfo(lae.Expr, col)
+		rt, foundR := ctx.SemTable.TypeForExpr(rae.Expr)
+		lt, foundL := ctx.SemTable.TypeForExpr(lae.Expr)
+		if foundR && foundL {
+			types := []sqltypes.Type{rt.Type(), lt.Type()}
+			t := evalengine.AggregateTypes(types)
+			ctx.SemTable.ExprTypes[col] = evalengine.NewType(t, collations.Unknown)
+		} else {
```

Comment:
I don't think we should have the else clause here. It's not correct to use either of the two types as the common type. Like my example showed, you can have a situation where one side is an int, the other side is a datetime, but the common type is varchar.

If we don't have both types available, I think we need to use the `weight_string` function to be sure we can compare values correctly.

Reply:
I understand now. There shouldn't be an "else" processing logic. 

If it is a syntax tree node for the now() function, we can save its corresponding ``evalengine.Type`` in ``ExprTypes`` during the analysis phase in the analyzer.